### PR TITLE
Dec 1254 child search

### PR DESCRIPTION
**Pull request by hbeachey**
Add the `../search/children` route to search only the children items of a collection.

By default, the resulting hits are returned grouped by the parent id. This can be turned off to return flat hits by adding the parameter `no_group=true` to the url.

### CODE REVIEW

File: spec/models/waggle/adapters/solr/search/result_spec.rb [hunk 1]
```diff
@@ -71,6 +71,39 @@
...
+        },
+      ).and_return("solr_response")
+      expect(subject.send(:connection)).to receive(:paginate).with(
+        2,
+        0,
+        "select",
+        params: {
+          q: "-part_parent_s:_is_parent_ AND a query",
+          group: true,
+          :"group.field" => "part_parent_s",
```
Comment (hbeachey) [inline spec/models/waggle/adapters/solr/search/result_spec.rb]: As far as I know, can't use this syntax with string keys. I was following the conventions set previously in these files.

File: app/controllers/v1/search_controller.rb [hunk 2]
```diff
@@ -13,6 +13,22 @@ def index
       @search = Waggle.search(**search_arguments)
     end
 
+    def children
```
Comment (jgondron) [inline app/controllers/v1/search_controller.rb]: Needs specs

File: app/models/waggle/adapters/solr/search/result.rb [hunk 3]
```diff
@@ -13,6 +13,10 @@ def hits
             @hits ||= solr_docs.map { |solr_doc| Waggle::Adapters::Solr::Search::Hit.new(solr_doc) }
           end
 
+          def groups
```
Comment (jgondron) [inline app/models/waggle/adapters/solr/search/result.rb]: Needs specs

File: app/views/v1/search/_group.json.jbuilder [hunk 4]
```diff
@@ -0,0 +1,4 @@
+json.groupId group.fetch(:id, "")
```
Comment (jgondron) [inline app/views/v1/search/_group.json.jbuilder]: The schema here could use a little work to look more like the rest of the objects. Right now, this produces:
```
{
  "groupId": "http://localhost:3017/v1/items/8e6a38769f",
  "hits": [
    {
      "@type": "SearchHit",
      "@id": "http://localhost:3017/v1/items/fad922ca22",
      "isPartOf/Item": "http://localhost:3017/v1/items/8e6a38769f",
      "type": "Metadata Only",
      "name": "CST Paragraph Level",
      "shortDescription": " ",
      "description": null,
      "thumbnailURL": null,
      "updated": "2017-01-11T18:51:51.000+00:00"
    }
  ]
}
```
Should perhaps be something more like:
```
{
  "@type": "SearchHitParent",
  "@id": "http://localhost:3017/v1/items/8e6a38769f",
  "children": [
    {
      "@type": "SearchHit",
      "@id": "http://localhost:3017/v1/items/fad922ca22",
      "isPartOf/Item": "http://localhost:3017/v1/items/8e6a38769f",
      "type": "Metadata Only",
      "name": "CST Paragraph Level",
      "shortDescription": " ",
      "description": null,
      "thumbnailURL": null,
      "updated": "2017-01-11T18:51:51.000+00:00"
    }
  ]
}
```

Comment (jonhartzler) [inline app/views/v1/search/_group.json.jbuilder]: Also consider using hits here so there is a consistent interface to the data.

File: app/controllers/v1/search_controller.rb (hunk 2 above)
Comment (hbeachey) [inline app/controllers/v1/search_controller.rb]: added

**Pull request merged**